packages/block-library/src/heading/edit.js

```diff
@@ -90,20 +92,22 @@ function HeadingEdit( {
 
 	return (
 		<>
-			<BlockControls group="block">
-				<HeadingLevelDropdown
-					value={ level }
-					onChange={ ( newLevel ) =>
-						setAttributes( { level: newLevel } )
-					}
-				/>
-				<AlignmentControl
-					value={ textAlign }
-					onChange={ ( nextAlign ) => {
-						setAttributes( { textAlign: nextAlign } );
-					} }
-				/>
-			</BlockControls>
+			{ blockEditingMode === 'default' && (
+				<BlockControls group="block">
+					<HeadingLevelDropdown
+						value={ level }
+						onChange={ ( newLevel ) =>
+							setAttributes( { level: newLevel } )
+						}
+					/>
```

Comment:
I disagree with the removal of this. Heading level is not a visual setting but rater a semantic one. Depending on where a pattern is used adjusting the heading level is required to make sure content follows good semantics and is as accessible as possible.

Reply:
True, though we'd need to explore making it work as an attribute that supports block binding, as it wasn't working. 

I guess there is still a question of whether it should be retained as viewable in contentOnly mode outside of patterns.

I'll add it to the tracking issue to explore.

Reply:
I would consider this a breaking change for how content only mode currently functions. 

I get there are implications for the new block connections and overrides.

But this is making the existing content only locking mode less usable.

Reply:
It was really a bug, as that attribute isn't marked `role: 'content'`, so shouldn't have been visible in 'contentOnly' mode from the start. The same with text align.

I understand the point that it limits important functionality though. I think it's one of those attributes that falls between the cracks in not being content, but being important for the structure of content.

I think there's some suffering from 'contentOnly' mode not being properly thought out, and personally I don't think the `role: 'content'` attributes were a good idea from the start. It's definitely an area that needs more work, and I think post 6.5 pattern overrides may lean on it less in favour of something a bit more expressive.